core/types/header_ext_test.go

```diff
@@ -242,3 +242,55 @@ func TestHeaderSerializable_updates(t *testing.T) {
 		assert.Equal(t, wantedExtras, extras)
 	})
 }
+
+func TestHeaderExtraPostCopy(t *testing.T) {
+	t.Parallel()
+
+	t.Run("empty_extra", func(t *testing.T) {
+		t.Parallel()
+
+		got := &ethtypes.Header{
+			ParentHash: common.Hash{1},
+		}
+		extra := &HeaderExtra{}
+
+		extra.PostCopy(got)
+
+		want := &ethtypes.Header{
+			ParentHash: common.Hash{1},
+		}
+		wantExtra := &HeaderExtra{}
+		_ = WithHeaderExtra(want, wantExtra)
+
+		assert.Equal(t, want, got)
+		gotExtra := GetHeaderExtra(got)
+		assert.Equal(t, wantExtra, gotExtra)
+	})
+
+	t.Run("filled_extra", func(t *testing.T) {
+		got := &ethtypes.Header{
+			ParentHash: common.Hash{1},
+		}
+		extra := &HeaderExtra{
+			ExtDataHash:    common.Hash{2},
+			ExtDataGasUsed: big.NewInt(3),
+			BlockGasCost:   big.NewInt(4),
+		}
+
+		extra.PostCopy(got)
```

Comment:
```suggestion
		got := CopyHeader(WithHeaderExtra(
			&ethtypes.Header{
				ParentHash: common.Hash{1},
			},
			&HeaderExtra{
				ExtDataHash:    common.Hash{2},
				ExtDataGasUsed: big.NewInt(3),
				BlockGasCost:   big.NewInt(4),
			},
		))
```

Same rationale as above.

Reply:
Test removed in favor of `TestCopyHeader`